docs/setup/yarn_setup.md

```diff
@@ -100,11 +100,13 @@ Usage:
      -D <arg>                        Dynamic properties
      -d,--detached                   Start detached
      -jm,--jobManagerMemory <arg>    Memory for JobManager Container [in MB]
+     -nm,--name                      Set a custom name for the application on YARN
```

Comment:
Your concern is justified. But, if somebody is doing `-n mycustomname` it will fail because mycustomname is not numeric ;)
The reason why I have the `-nm` argument as well is because the user who requested the feature suggested to use exactly that name ;)